### ISSUE TITLE
A 500 error is raised when user change the object ID during subscription

### ISSUE THREAD

**Issue by ncrocfer** [opencve/opencve#121]
## Expected Behavior
Handle a bad object ID.

## Actual Behavior
If a user changes the ID of a vendor or a product ID during subscription a 500 response is raised.

## Steps to Reproduce the Problem

  - list the CVEs of a vendor or a product
  - open the browser inspector
  - change the ID of the vendor or product
  - click on the subscribe button

## Specifications

  - OpenCVE version: 1.20
  - Python version: 3.7.1

## Screenshots

![opencve_issue_subscribe](https://user-images.githubusercontent.com/1552526/125524663-ff44da1e-81d5-4593-a7c5-93e4d368a3ce.png)
![opencve_issue_subscribe_2](https://user-images.githubusercontent.com/1552526/125524673-1cacf67a-03a0-4aee-8651-490ed8efbd2d.png)

##

**Comment by zachw66** [comment]
I'm currently working on this issue and one thing that I'm realizing is that this is catching the error when I test it with a POST request and is returning the json properly: 
```
       try:
            vendor = Vendor.query.get(request.form["id"])
        except exc.SQLAlchemyError:
            return make_response(jsonify({"status": "error", "message": "vendor " + request.form["id"] + " does not exist"}), 404)
```

But when I actually boot up the application and test it, it's somehow continuing past that and to catch it entirely I need to add this:
```
       if str(vendor) == "None":
            return make_response(jsonify({"status": "error", "message": "vendor " + request.form["id"] + " does not exist"}), 404)
```

Now, both of these code blocks are necessary for catching the error (if I remove the first one, it still throws the 500 error on the application), but I'm not sure if I'm missing something with my code and could be catching it all at once. I find it odd that my tests are getting a different response and behaving differently than the application. @ncrocfer I know you probably have quite a bit of SQLAlchemy knowledge, so if you have some insight here I would appreciate it. If you think having the two-part catch with the try-catch and the if statement is good enough as it catches the problem, I can make the PR.

Additionally, would we like to have an element added to the UI when this error occurs? We could just add something to the AJAX if it receives an error.

**Comment by ncrocfer** [comment]
Reuse what's already exist, so just create a function returning the response:

```python
@main.route("/subscriptions", methods=["POST"])
@login_required
def subscribe_to_tag():
    def _bad_request():
        return jsonify({"status": "error", "message": "bad request"}), 400
```

And use the `opencve.models.is_valid_uuid` and the `VendorController.get` functions, something like:

```python
if request.form["obj"] == "vendor":
        if not is_valid_uuid(request.form["id"]):
            return _bad_request()
        vendor = VendorController.get({"id": request.form["id"]})
```

You can customize the messages of bad requests if you want, but please don't use a concatenation but f-strings.

**Comment by zachw66** [comment]
@ncrocfer Just realizing this issue hasn't been closed yet. Feel free to do that whenever you get the chance. 😃

**Comment by ncrocfer** [comment]
Fixed by #126
